server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -696,6 +696,46 @@ static String getCommitCheckpointFileName(long generation) {
         return TRANSLOG_FILE_PREFIX + generation + CHECKPOINT_SUFFIX;
     }
 
+    /**
+     * Trims translog (effectively moves max visible seq# {@link Checkpoint#trimmedAboveSeqNo}) for terms below <code>belowTerm</code>
```

Comment:
also we should be clear we're talking about primary terms of the files not the operations.